messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingExtension.java

```diff
@@ -167,9 +168,10 @@ public static ResourceDescriptionResolver getResourceDescriptionResolver(final b
         return new StandardResourceDescriptionResolver(prefix.toString(), RESOURCE_NAME, MessagingExtension.class.getClassLoader(), true, useUnprefixedChildTypes);
     }
 
+    @Override
     public void initialize(ExtensionContext context) {
         final SubsystemRegistration subsystemRegistration = context.registerSubsystem(SUBSYSTEM_NAME, CURRENT_MODEL_VERSION);
-        subsystemRegistration.registerXMLElementWriter(new MessagingSubsystemParser_2_0());
+        subsystemRegistration.registerXMLElementWriter(MessagingSubsystemParser_3_0::new);
```

Comment:
Could you create a static instance for the parser and use it both to here and in `initializeParsers`?

Reply:
The registerXMLElementWriter(...) method that accepts an XMLElementWriter instance is deprecated.  If this is OK to use now, why is it deprecated?

Reply:
Nevermind - I see that the deprecation annotation was removed in wildfly-core master.